rdt/transformers/numerical.py

```diff
@@ -188,13 +188,14 @@ def _reverse_transform(self, data):
             min_bound, max_bound = INTEGER_BOUNDS[self.computer_representation]
             data = data.clip(min_bound, max_bound)
 
-        is_integer = np.dtype(self._dtype).kind == 'i'
+        is_integer = pd.api.types.is_integer_dtype(self._dtype)
+        is_pandas_instance = isinstance(data, (pd.Series, pd.DataFrame))
         if self.learn_rounding_scheme and self._rounding_digits is not None:
             data = data.round(self._rounding_digits)
         elif is_integer:
             data = data.round(0)
 
-        if pd.isna(data).any() and is_integer:
+        if pd.isna(data).any() and is_integer and not is_pandas_instance:
```

Comment:
The new pandas dtype support NaN integers so we could remove this if statement. However I had to add the check for pandas instance to make this test work:
https://github.com/sdv-dev/RDT/blob/dd654c4911e13f95a6afe70d4d34f4013bd4f237/tests/unit/transformers/test_numerical.py#L461
Let me know if this is fine. All our transformers accept `pandas` and `np.ndarray` object right?

Reply:
I think we will have to revisit `RDT` on a performance audit against large inputs. The ideal scenario is that we have a `pd.DataFrame` as input (on the public method), and here we work with `pd.Series` or `np.array`, what it would be ideal is to work only with one input type and avoid the overhead of supporting two types of inputs.